upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
+        const end = Math.min(start + chunkSize, file.size);
+        const chunk = file.slice(start, end);
+
+        // Creating a new XMLHttpRequest for every chunk (bad practice)
+        var xhr = new XMLHttpRequest();
+
+        // Opening a new connection every time inside the loop (also not great)
+        xhr.open("POST", serverUrl, true);
+
```

Comment:
Setting request headers within the loop is redundant. They should be set once outside the loop if they do not change across requests.